sp/src/game/server/props.cpp

```diff
@@ -5388,34 +5391,54 @@ bool CBasePropDoor::HandleInteraction( int interactionType, void *data, CBaseCom
 {
 	if ( interactionType == g_interactionBadCopKick)
 	{
-		// Fire an output
-		m_OnKicked.FireOutput( sourceEnt, this, 0.0f );
+		bool kicked = KickOpen( sourceEnt );
 
-		// If kicking is meant to unlock this door, unlock
-		if ( m_bUnlockOnKick )
-		{
-			Unlock();
-		}
+		// Set success if we opened
+		KickInfo_t *info = static_cast<KickInfo_t*>(data);
+		if (info)
+			info->success = kicked;
 
-		if ( CanOpenOnKick( sourceEnt ) )
-		{
-			// Set the door speed to the kicking speed
-			m_bKicked = true;
-
-			// Open the door away from the source entity if you can
-			OpenIfUnlocked( sourceEnt, sourceEnt );
-
-			// Return successful if we're opening
-			KickInfo_t *info = static_cast<KickInfo_t*>(data);
-			if (info)
-				info->success = IsDoorOpening();
-		}
+		g_EventQueue.AddEvent( STRING( GetEntityName() ), "KickOpen", variant_t(), 0.0f, this, this, 0 );
```

Comment:
Oops - the activator needs to be ``sourceEnt``, not ``this``! Otherwise this will not work!